### DISCUSSION TITLE
How to use custom views of the layout.

### DISCUSSION THREAD

**Discussion by Usama99786** [AppIntro/AppIntro#1009] [Q&A] [answered]
I want to navigate through the slides using my own custom views .i.e buttons etc. I want to hide the default next button and use my own views. Is there a way to do that?

**Comment by cortinico** [comment, marked as answer]
Yes that's doable. You'll have to create your own `AppIntroBase` subclass:

https://github.com/AppIntro/AppIntro/blob/e635ec9f91c9703461a4e137f3327315afc3d9a9/appintro/src/main/java/com/github/appintro/AppIntro2.kt#L12-L14

Your class will have to expose an `layoutId` that will be used by AppIntro to populate the layout.
You'll also have to make sure the IDs of your views are the same as `app_intro_layout`.

I'd suggest you copy over this file and start over from there customizing your setup:
https://github.com/AppIntro/AppIntro/blob/main/appintro/src/main/res/layout/appintro_intro_layout.xml

**Comment by kamathprasad9** [comment]
@Usama99786 were you able to achieve custom views? If yes, could you provide some snippets for my reference? Thank you!
